### PR TITLE
r/ses_event_destination - add arn + validation

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #13624

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_ses_event_destination: add arn attribute
resource_aws_ses_event_destination: plan time validation for `name`, `cloudwatch_destination.default_value`, `cloudwatch_destination.default_name`, `kinesis_destination.role_arn`, `kinesis_destination.stream_arn`,
`sns_destination.topic_arn`
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSSESEventDestination_'
--- PASS: TestAccAWSSESEventDestination_basic (179.55s)
--- PASS: TestAccAWSSESEventDestination_disappears (160.52s)
```

### CODE REVIEW

**Comment by bflad** [comment]
Ready for rebase and changelog (can do myself later today if you're not available 👍 )

**Comment by DrFaust92** [comment]
On it

File: .changelog/13964.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+```release-note:enhancement
+resource/ses_event_destination: Add `arn` attribute.
```
Comment (bflad) [inline .changelog/13964.txt:2]:
```suggestion
resource/aws_ses_event_destination: Add `arn` attribute
```

File: .changelog/13964.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+```
+
+```release-note:enhancement
+resource/ses_event_destination: Add plan time validation for `name`, `cloudwatch_destination.default_value`, `cloudwatch_destination.default_name`, `kinesis_destination.role_arn`, `kinesis_destination.stream_arn`,
+`sns_destination.topic_arn`.
```
Comment (bflad) [inline .changelog/13964.txt:7]:
```suggestion
resource/aws_ses_event_destination: Add plan time validation for `name`, `cloudwatch_destination.default_value`, `cloudwatch_destination.default_name`, `kinesis_destination.role_arn`, `kinesis_destination.stream_arn`, and `sns_destination.topic_arn` attributes
```

File: website/docs/r/ses_event_destination.html.markdown [hunk 3]
```diff
@@ -89,6 +89,14 @@ The following arguments are supported:
...
 * `topic_arn` - (Required) The ARN of the SNS topic
 
+## Attributes Reference
+
+In addition to the arguments, which are exported, the following attributes are exported:
+
+* `id` - The SES event destination name.
+* `arn` - The SES event destination ARN.
+
+
```
Comment (bflad) [inline website/docs/r/ses_event_destination.html.markdown:99]: 😎  extra newline

```suggestion
```

**Review by bflad — APPROVED** [review]
Excellent, looks good to me, @DrFaust92 🚀 

References:

- https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazonses.html#amazonses-resources-for-iam-policies
- https://docs.aws.amazon.com/ses/latest/APIReference/API_EventDestination.html

Output from acceptance testing in AWS Commercial:

```
--- PASS: TestAccAWSSESEventDestination_disappears (85.94s)
--- PASS: TestAccAWSSESEventDestination_basic (98.02s)
```

Output from acceptance testing in AWS GovCloud (US):

```
--- PASS: TestAccAWSSESEventDestination_disappears (75.67s)
--- PASS: TestAccAWSSESEventDestination_basic (78.60s)
```

**Pull request merged**